toolkit/tools/pkg/imagecustomizerlib/customizeuki.go

```diff
@@ -59,18 +65,67 @@ type UkiKernelInfo struct {
 	Initramfs string `json:"initramfs"`
 }
 
-func prepareUki(ctx context.Context, buildDir string, uki *imagecustomizerapi.Uki, imageChroot *safechroot.Chroot,
+func baseImageHasUkis(imageChroot *safechroot.Chroot) (bool, error) {
+	espDir := filepath.Join(imageChroot.RootDir(), EspDir)
+	ukiFiles, err := getUkiFiles(espDir)
+	if err != nil {
+		return false, fmt.Errorf("failed to check for UKI files:\n%w", err)
+	}
+	return len(ukiFiles) > 0, nil
+}
+
+// validateUkiReinitialize validates the UKI reinitialize mode against the base image state.
+// Rules:
+// - If base image has UKIs and reinitialize is unspecified (empty string), return an error
+// - If base image has UKIs and reinitialize is "passthrough", allow (preserve existing UKIs)
+// - If base image has UKIs and reinitialize is "refresh", allow (regenerate UKIs)
+// - If base image has no UKIs, no validation needed (config.OS.Uki can be nil or any value)
+func validateUkiReinitialize(imageConnection *imageconnection.ImageConnection, config *imagecustomizerapi.Config) error {
+	hasUkis, err := baseImageHasUkis(imageConnection.Chroot())
+	if err != nil {
+		return err
+	}
+
+	if !hasUkis {
+		// Base image doesn't have UKIs, no validation needed
+		return nil
+	}
+
+	// Base image has UKIs
+	if config.OS == nil || config.OS.Uki == nil {
+		return fmt.Errorf("base image contains UKI files but config does not specify os.uki field: " +
```

Comment:
We don't need two separate error messages for this. Just use the one below for both cases.

Reply:
Fixed.